apps/surfer/src/hooks/useContentHtml.ts

```diff
@@ -0,0 +1,31 @@
+import { EntryFieldAPI } from '@contentful/app-sdk';
+import { documentToHtmlString } from '@contentful/rich-text-html-renderer';
+import { useEffect, useMemo, useState } from 'react';
+
+const getAllHtml = (fieldValues: any[]) => fieldValues.map((field) => documentToHtmlString(field)).join('');
+
+export const useContentHtml = (allFields: EntryFieldAPI[], selectedFields: EntryFieldAPI[]) => {
+  const [contentHtml, setContentHtml] = useState<string>('');
+  const [fieldValues, setFieldValues] = useState<{ [fieldId: string]: string }>({});
+
+  const selectedFieldsIds = useMemo(() => selectedFields.map(({ id }) => id), [selectedFields]);
+
+  useEffect(() => {
+    for (const field of allFields) {
+      field.onValueChanged((value) => {
+        setFieldValues((values) => ({ ...values, [field.id]: value }));
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
`allFields` dep breaks this?

Reply:
Yup. And apart from that, I only want to run this once anyway.